src/Raven.Server/Documents/ETL/Providers/Raven/Enumerators/CountersToRavenEtlItems.cs

```diff
@@ -1,26 +1,39 @@
 ﻿using System.Collections;
 using System.Collections.Generic;
 using Raven.Client.Documents.Operations.Counters;
+using Raven.Server.ServerWide.Context;
 
 namespace Raven.Server.Documents.ETL.Providers.Raven.Enumerators
 {
     public class CountersToRavenEtlItems : IEnumerator<RavenEtlItem>
     {
+        private readonly DocumentsOperationContext _context;
         private readonly IEnumerator<CounterGroupDetail> _counters;
         private readonly string _collection;
 
-        public CountersToRavenEtlItems(IEnumerator<CounterGroupDetail> counters, string collection)
+        public CountersToRavenEtlItems(DocumentsOperationContext context, IEnumerator<CounterGroupDetail> counters, string collection)
         {
+            _context = context;
             _counters = counters;
             _collection = collection;
         }
 
+        private bool Filter(RavenEtlItem item)
+        {
+            var doc = _context.DocumentDatabase.DocumentsStorage.Get(_context, item.DocumentId, DocumentFields.Default);
+
+            // counter has lower etag than its document - we skip it to avoid
+            // sending a counter of document that can not exist on the destination side
```

Comment:
what will send the counter to the other side afterward? won't this cause us to miss a write?

Reply:
we always load the counters together with the document itself.
https://github.com/ravendb/ravendb/blob/4569e8a7bf34cc94c722b85bd16e6f342d3df663/src/Raven.Server/Documents/ETL/Providers/Raven/RavenEtlDocumentTransformer.cs#L217